libs/src/sdk/types/HashId.ts

```diff
@@ -0,0 +1,10 @@
+import { z } from 'zod'
+import { decodeHashId } from '../utils/hashId'
+
+export const HashId = z.string().transform<number>((data: string) => {
+  const id = decodeHashId(data)
+  if (id === null) {
+    throw new Error('id is not valid')
```

Comment:
Prob better to have a more informative/formal error message in case this gets surfaced to the consumer